DEVEL.md

```diff
@@ -0,0 +1,78 @@
+# Developing the Maistra Istio Operator
+
+## Building the binary and the container image
+
+To build the Operator container image, run:
+```bash
+IMAGE=docker.io/<your username>/istio-operator:latest tmp/build/docker_build.sh
+```
+
+This builds the binary, downloads and patches the Istio Helm charts, and finally builds the container image. The build script doesn't push the image to the container registry, so you'll need to push it manually:
+```bash
+docker push docker.io/<your username>/istio-operator:latest
+```
+
+## Running the operator locally
+
+During development, you can run Istio Operator locally on your dev machine and have it reconcile objects in a local or remote OpenShift cluster. You can run the Operator from within your IDE and even use a debugger.
+
+To run the Operator locally, you need to set the following environment variables:
+
+- `KUBECONFIG=<path to your kubectl config file>`
+- `POD_NAME=istio-operator`
+- `POD_NAMESPACE=istio-operator`
+- `ISTIO_CNI_IMAGE=some-registry/istio-cni-image:latest`
+- `ISTIO_CNI_IMAGE_PULL_SECRET=my-pull-secret` (required if the registry hosting the CNI image is not public; secret must be in operator's namespace as defined above)
+
+Make sure you run `tmp/build/docker_build.sh` before running the Operator to generate the Helm charts in `tmp/_output/helm`.
+
+Before running the Operator, don't forget to stop the Operator running in the cluster:
+```bash
+oc -n istio-operator scale deployment istio-operator --replicas 0
+```  
+
+Run the operator locally with the following command-line options:
+```bash
+--resourceDir tmp/_output/resources/ --discoveryCacheDir /tmp/istio-operator/.kube/cache/discovery istio-operator
+```
+
+### Modifying the ValidatingWebhookConfiguration 
+
+Because the Operator installs a `ValidatingWebhookConfiguration` that points to the Operator pod running inside the cluster, all CRUD operations on `ServiceMeshControlPlane` or `ServiceMeshMemberRoll` objects will fail. To fix this, you either need to remove the `ValidatingWebhookConfiguration`:
+```bash
+oc delete validatingwebhookconfigurations istio-operator.servicemesh-resources.maistra.io 
```

Comment:
It gets recreated by the operator (via code).

So we get errors like
```
ERROR	kubebuilder.controller	controller/controller.go:209	Reconciler error	{"Controller": "servicemeshmemberroll-controller", "Request": "i1/default", "error": "Internal error occurred: failed calling webhook \"smmr.validation.maistra.io\": Post https://admission-controller.istio-operator.svc:443/validate-smmr?timeout=30s: no endpoints available for service \"admission-controller\""}
```

Reply:
Yes, I think I mention that you need to delete it every time you run the operator.

Reply:
Ah ok, gotcha. I missed an **after** in the sentence ... every time you restart the Operator...